rebasehelper/config.py

```diff
@@ -0,0 +1,83 @@
+# -*- coding: utf-8 -*-
+#
+# This tool helps you to rebase package to the latest version
+# Copyright (C) 2013-2014 Red Hat, Inc.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# he Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+# Authors: Petr Hracek <phracek@redhat.com>
+#          Tomas Hozza <thozza@redhat.com>
+
+import configparser
+import os.path
+
+from rebasehelper.logger import logger
+from rebasehelper.options import OPTIONS
+
+
+class Conf:
+    def __init__(self, cli):
+        self.cli = cli
+        self.path_to_config = self.get_conf_path()
+        self.config = self.get_config()
+
+    def get_conf_path(self):
+        def get_conf_option():
+            for option in OPTIONS:
+                if "name" in option.keys():
+                    for item in option["name"]:
+                        if item == "--conf":
+                            return option
+
+        if "conf" in vars(self.cli.args).keys():
+            path_to_conf = os.path.abspath(self.cli.args.conf)
+        else:
+            conf_option = get_conf_option()
+            path_to_conf = conf_option["default"]
+        return path_to_conf
+
+    def get_config(self):
+        conf = {}
+        if os.path.isfile(self.path_to_config):
+            config = configparser.ConfigParser()
+            config.read(self.path_to_config)
+            for section in config.sections():
+                for key, val in config.items(section):
+                    dest = key.replace("-", "_")
+                    conf[dest] = val
+        else:
+            logger.info('Could not find any set or default config file')
+
+        return conf
+
+    def merge(self):
+        for key, value in vars(self.cli.args).items():
+            self.config[key] = value
+
+        for option in OPTIONS:
+            if "dest" in option.keys():
+                dest = option["dest"]
+            else:
+                dest = max(option["name"], key=len)
+                dest = dest.lstrip("-").replace("-", "_")
+
+            if dest not in self.config.keys():
+                self.config[dest] = option.get("default")
+
+    def __getattr__(self, name):
+        try:
+            self.config[name]
```

Comment:
There is no `return` here.

Reply:
And you don't have to catch `KeyError`, `self.config.get(name)` will handle it for you.